File: core/src/test/java/de/qabel/core/config/IdentityTestFactory.java
package de.qabel.core.config;

import de.qabel.core.crypto.QblEcPairTestFactory;
import org.meanbean.lang.Factory;

/**
 * IdentityTestFactory
 * Creates distinct instances of class Identity
 * Attention: For testing purposes only
 */
public class IdentityTestFactory implements Factory<Identity> {
    DropUrlListTestFactory urlListFactory;
    QblEcPairTestFactory qpkpFactory;

    public IdentityTestFactory() {
        urlListFactory = new DropUrlListTestFactory();
        qpkpFactory = new QblEcPairTestFactory();
    }

    @Override
    public Identity create() {
        return new Identity("alias", urlListFactory.create(), qpkpFactory.create());
    }
}

File: core/src/test/java/de/qabel/core/config/IdentityTestFactory.kt
package de.qabel.core.config

import de.qabel.core.crypto.QblEcPairTestFactory
import org.meanbean.lang.Factory

/**
 * IdentityTestFactory
 * Creates distinct instances of class Identity
 * Attention: For testing purposes only
 */
class IdentityTestFactory : Factory<Identity> {
    internal var urlListFactory: DropUrlListTestFactory
    internal var qpkpFactory: QblEcPairTestFactory

    init {
        urlListFactory = DropUrlListTestFactory()
        qpkpFactory = QblEcPairTestFactory()
    }

    override fun create(): Identity {
        return Identity("alias", urlListFactory.create(), qpkpFactory.create())
    }
}
